File: 백준/Silver/3986. 좋은 단어/좋은 단어.cc
#include<bits/stdc++.h> 
using namespace std;  
int n, ret; 
string s; 
int main() {
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);
	cin >> n; 
	for(int i = 0; i < n; i++){
		cin >> s; 
		stack<char> stk;
		for(char a : s){
			if(stk.size() && stk.top() == a)stk.pop();
			else stk.push(a); 
		} 
		if(stk.size() == 0)ret++;
	}
	cout << ret << "\n"; 
}

File: 백준/Silver/3986. 좋은 단어/좋은 단어.py
n = int(input())
ret = 0
for _ in range(n):
    s = input()
    stk = []
    for c in s:
        if stk and stk[-1] == c:
            stk.pop()
        else: stk.append(c)
    if stk: continue
    ret += 1
print(ret)